docs/tutorial.md

```diff
@@ -0,0 +1,231 @@
+  Csv,
+  SaveMode.ErrorIfExists,
+  options = Map("header" -> "true", "inferSchema" -> "true"))
+``` 
+
+This is telling Lighthouse: You will find a csv file in the folder `data/airplane`. The csv file has headers. You can infer the schema yourself. Now, we need to have a unique way to refer to this `DataLink`. The way we do that is by linking the `DataLink` to a `DataUID` which has 2 parameters: a `namespace` and a `key`. You can think of the namespace as the zone of your data lake where this data belongs, and the key as the actual name of the `DataLink`. So for this raw airplane data, we could define the `DataUID` as follows:
+
+```scala
+val dataUID = DataUID("raw", "airplane")
+```
+This is telling Lighthouse: I have an `airplane` dataset, in the `raw` zone of the data lake. 
+All data sources coming from the outside world are considered `raw`. Which means it can be any format: csv, json, avro, gzip, ... Later on, we will transform those data sources into a structured format, by default `ORC`.
+
+Finally, we are ready to define a simple test environment with that one data source:
+```scala
+ environment("test") { refs =>
+    refs += DataUID("raw", "airplane") -> new FileSystemDataLink(
+      resource("data/airplane"),
+      Csv,
+      SaveMode.ErrorIfExists,
+      options = Map("header" -> "true", "inferSchema" -> "true"))
+ }
+```
+Congratulations, you have built your first data lake. It has one environment: `test`, which contains one `DataLink`: a csv file loaded from your local disk. You cataloged that data to be in the `raw` zone of your data lake and it is named `airplane`. Let's extend this with two weather sources, in a similar fashion, and wrap it all in our very own `AirplaneDatalake` object.
+
+```scala
+object AirplaneDatalake extends Datalake {
+     environment("test") { refs =>
+        refs += DataUID("raw", "airplane") -> new FileSystemDataLink(
+          resource("data/airplane"),
+          Csv,
+          SaveMode.ErrorIfExists,
+          options = Map("header" -> "true", "inferSchema" -> "true"))
+          
+        refs += DataUID("raw.weather", "daily") -> new FileSystemDataLink(
+            resource("data/weather/daily"),
+            Csv,
+            SaveMode.ErrorIfExists,
+            options = Map("header" -> "true", "inferSchema" -> "true"))
+        
+        refs += DataUID("raw.weather", "station") -> new FileSystemDataLink(
+            resource("data/weather/station"),
+            Csv,
+            SaveMode.ErrorIfExists,
+            options = Map(
+              "header" -> "true", 
+              "inferSchema" -> "true", 
+              "delimiter"   -> "|"))
+     }
+     
+     private def resource(path: String): String = File.resource(path).pathAsString
+ }
+``` 
+
+Great, now how can you use this? Well, from this moment onwards, you can always access the data sources by just asking the data lake:
+
+```scala
+    val df = AirplaneDatalake("raw" -> "airplane").read()
```

Comment:
Maybe mention the DataUID version as well as this shows that there is a direct link between the DataUID and the data